File: src/Game.hs

-- Copyright (c) 2014, 2015 Mario Mlačak, mmlacak@gmail.com
-- Licensed under 3-clause (modified) BSD license. See LICENSE.txt for details.

module Game
where

import qualified BoardType as BT
import qualified Rules as R


data Game = Game { rules :: R.Rules }
-- TODO :: add aI :: (Maybe ?) Ai
    deriving (Show, Read)


initializeGame :: BT.BoardType -> Game
initializeGame bt = Game { rules=(R.initializeRules bt) }

File: book/py/game.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

# Copyright (c) 2010, .. 2016 Mario Mlačak, mmlacak@gmail.com
# Licensed under 3-clause (modified) BSD license. See LICENSE.txt for details.

from rules import Rules

class Game(object):
    def __init__(self, rules):
        self.rules = rules

    # TODO :: this is where fusion with Ai and user responses comes
